pkg/providers/instance/instance.go

```diff
@@ -353,6 +356,15 @@ func (p *Provider) updateUnavailableOfferingsCache(ctx context.Context, errors [
 	for _, err := range errors {
 		if awserrors.IsUnfulfillableCapacity(err) {
 			p.unavailableOfferings.MarkUnavailableForFleetErr(ctx, err, capacityType)
+			// Add a k8s event for the instance type and zone without the involved object which has an ICE error
+			instanceType := aws.StringValue(err.LaunchTemplateAndOverrides.Overrides.InstanceType)
+			availabilityZone := aws.StringValue(err.LaunchTemplateAndOverrides.Overrides.AvailabilityZone)
+			p.recorder.Publish(events.Event{
```

Comment:
Could we create a function to generate this event (perhaps in an `events.go` file based on some parameters so other callers that might generate this error don't have to know all these details?

Reply:
Hello @jonathan-innis , thanks your review, I tried addressing your comment in the next commit, please have a look, thank you.

Reply:
Hello @Jonathan Innis, I updated the PR to move the event being published inside when updating the cache for an instancetype, zone, capacity type and combination, please have a look whenever you get time. https://github.com/aws/karpenter/pull/3999